tests/ext/sandbox/install_hook/trace_closure_ge_php_84.phpt

```diff
@@ -0,0 +1,101 @@
+--TEST--
+Tracing Closures via install_hook()
+--SKIPIF--
+<?php
+if (PHP_VERSION_ID < 80400) {
+    die('skip: test only stable on PHP >= 8.4');
+}
+?>
+--INI--
+datadog.trace.generate_root_span=0
+--FILE--
+<?php
+
+namespace test;
+
+$internalFakeClosure = (new \ReflectionFunction("intval"))->getClosure();
+
+$topLevelClosure = function($a) {
+    return $a + 1;
+};
+
+function foo() {
+    return function($a) {
+        return $a + 2;
+    };
+}
+
+class bar {
+    static function foo() {
+        return function($a) {
+            return $a + 3;
+        };
+    }
+}
+
+$closures = [$internalFakeClosure, $topLevelClosure, foo(), bar::foo()];
+$hooks = [];
+
+foreach ($closures as $i => $closure) {
+    $hooks[] = \DDTrace\install_hook($closure, function(\DDTrace\HookData $hook) use ($i) {
+        $hook->span()->resource = $i;
+    }, function(\DDTrace\HookData $hook) {
+        $hook->span()->meta['result'] = $hook->returned;
+    }, \DDTrace\HOOK_INSTANCE);
+
+    $closure(0);
+}
+
+foo()(0); // Not traced
+
+// Still traced
+foreach ($closures as $closure) {
+    $closure(1);
+}
+
+foreach ($hooks as $hook) {
+    \DDTrace\remove_hook($hook);
+}
+
+// Not traced
+foreach ($closures as $closure) {
+    $closure(2);
+}
+
+include __DIR__ . '/../dd_dumper.inc';
+\dd_dump_spans(true);
+
+?>
+--EXPECTF--
+spans(\DDTrace\SpanData) (8) {
+  intval (trace_closure_ge_php_84.php, 0, cli)
+    result => 0
+    _dd.p.tid => %s
+  {closure:%s.php:7\{closure} (trace_closure_ge_php_84.php, 1, cli)
```

Comment:
@bwoebi this does not seem right, there is a missing closing bracket
```diff
- {closure:%s.php:7\{closure}
+ {closure:%s.php:7\{closure\}}
```
should maybe look something like this?

Reply:
Why write "{closure}" twice? I think it should look like `{closure:%s.php:7}`, which is the format PHP 8.4 also uses for closures now.

Reply:
Sorry, I mean that 
```
  {closure:%s.php:7\{closure} (trace_closure_ge_php_84.php, 1, cli)
```
is the version that makes the test succeed, so I am thinking we might need some changes in the extension.

Reply:
Yes, we do :-)